File: 06 arrays/exercise - moveZeroes.py
# 283. Move Zeroes
# https://leetcode.com/problems/move-zeroes/description/

class Solution(object):
  def moveZeroes(self, nums):
    for i in reversed(range(len(nums) - 1)):
      if (0 == nums[i]):
        for j in range(i+1, len(nums)):
          nums[j-1] = nums[j]
        nums[-1] = 0

File: 06 arrays/exercise - moveZeroes.cs
// 283. Move Zeroes
// https://leetcode.com/problems/move-zeroes/description/

public class Solution {
    public void MoveZeroes(int[] nums) {
        for (int i = nums.Length - 2; i >= 0; i--)
        {
          if (nums[i] == 0)
          {
            for (int j = i + 1; j < nums.Length; j++)
            {
              nums[j - 1] = nums[j];
            }
            nums[nums.Length - 1] = 0;
          }
        }
    }
}